### ISSUE TITLE
RelAlgExecutor.cpp:1177 Check failed in work at MODIN_STORAGE_FORMAT=omnisci

### ISSUE THREAD

**Issue by Rubtsowa** [modin-project/modin#3788]
When I run example at MODIN_STORAGE_FORMAT=omnisci:

```
import modin.pandas as pd
frame_data = np.random.randint(0, 100, size=(2 ** 10, 2 ** 6))
df = pd.DataFrame(frame_data).add_prefix("col")
modin_dict = {"c": df.copy(), "b": df.copy()}
modin_result = pd.concat(modin_dict.values(), keys=modin_dict.keys())
```

I have ``Aborted (core dumped)``:
```
2021-12-07T07:56:30.157184 F 4029579 0 2 RelAlgExecutor.cpp:1177 Check failed: it_ok.second
Fatal Python error: Aborted
```

Similar error on startup:
```
import modin.pandas as pd
modin_series = pd.Series(list(range(1000)))
hasattr(pd.concat([modin_series, modin_series]), "index")
```

Also next modin tests have this error:
```
modin/pandas/test/test_series.py::test_append
modin/pandas/test/dataframe/test_binary.py::test_mismatched_row_partitions
modin/pandas/test/dataframe/test_udf.py::test_pipe
```

**Comment by vnlitvinov** [comment]
@modin-project/modin-omnisci could someone please check if this still happens?

**Comment by AndreyPavlenko** [comment]
Only the following test fails for me:
modin/pandas/test/dataframe/test_udf.py::test_pipe

    def drop(self, index=None, columns=None):
>       assert index is None, "Only column drop is supported"
E       AssertionError: Only column drop is supported

modin/experimental/core/storage_formats/omnisci/query_compiler.py:554: AssertionError

**Comment by AndreyPavlenko** [comment]
The above error has been fixed by #4100. All tests pass now. Closing the issue.
